client/my-sites/patterns/components/pattern-preview/index.tsx

```diff
@@ -54,48 +59,73 @@ export function PatternPreview( {
 		};
 	}, [ isCopied ] );
 
+	useEffect( () => {
+		authModalWrapperRef.current = document.createElement( 'div' );
+		document.body.appendChild( authModalWrapperRef.current );
+
+		return () => {
+			if ( authModalWrapperRef.current ) {
+				document.body.removeChild( authModalWrapperRef.current );
+				authModalWrapperRef.current = null;
+			}
+		};
+	}, [] );
+
 	if ( ! pattern ) {
 		return null;
 	}
 
 	return (
-		<div
-			className={ classNames( 'pattern-preview', className, {
-				'is-loading': ! renderedPattern,
-			} ) }
-		>
-			{ resizeObserver }
-
-			<div className="pattern-preview__renderer">
-				<PatternRenderer
-					minHeight={ nodeSize.width ? nodeSize.width / ASPECT_RATIO : undefined }
-					patternId={ patternId }
-					viewportWidth={ viewportWidth }
-				/>
-			</div>
+		<>
+			<div
+				className={ classNames( 'pattern-preview', className, {
+					'is-loading': ! renderedPattern,
+				} ) }
+			>
+				{ resizeObserver }
 
-			<div className="pattern-preview__header">
-				<div className="pattern-preview__title">{ pattern.title }</div>
-
-				{ canCopy && (
-					<ClipboardButton
-						className="pattern-preview__copy"
-						onCopy={ () => {
-							setIsCopied( true );
-						} }
-						text={ pattern?.html ?? '' }
-						primary
-					>
-						{ copyButtonText }
-					</ClipboardButton>
-				) }
+				<div className="pattern-preview__renderer">
+					<PatternRenderer
+						minHeight={ nodeSize.width ? nodeSize.width / ASPECT_RATIO : undefined }
+						patternId={ patternId }
+						viewportWidth={ viewportWidth }
+					/>
+				</div>
 
-				{ ! canCopy && (
-					<Button className="pattern-preview__get-access" transparent>
-						<Icon height={ 18 } icon={ lock } width={ 18 } /> Get access
-					</Button>
-				) }
+				<div className="pattern-preview__header">
+					<div className="pattern-preview__title">{ pattern.title }</div>
+
+					{ canCopy && (
+						<ClipboardButton
+							className="pattern-preview__copy"
+							onCopy={ () => {
+								setIsCopied( true );
+							} }
+							text={ pattern?.html ?? '' }
+							primary
+						>
+							{ copyButtonText }
+						</ClipboardButton>
+					) }
+
+					{ ! canCopy && (
+						<Button
+							className="pattern-preview__get-access"
+							transparent
+							onClick={ () => setIsAuthModalOpen( true ) }
+						>
+							<Icon height={ 18 } icon={ lock } width={ 18 } /> Get access
+						</Button>
+					) }
+				</div>
 			</div>
-		</div>
+
+			{ isAuthModalOpen &&
+				authModalWrapperRef.current &&
+				ReactDOM.createPortal(
+					<PatternsGetAccessModal onClose={ () => setIsAuthModalOpen( false ) } />,
+					authModalWrapperRef.current!
+				) }
```

Comment:
Did you consider using `react-modal`?

Reply:
With react-modal it's not necessary to create wrapper-node for modal, it's really cool, but I prefer to have more control over code.
I would go with custom (current) solution, but I totally don't mind to rewrite to react-modal if you have some preferences.

Reply:
`@automattic/components` has the `Dialog` component, which builds on `react-modal`. I'd argue that we should use that one instead of a custom solution. That way, we get better accessibility, a click-to-close background, scroll locking, <kbd>Esc</kbd> support etc.